core/cautils/getter/loadpolicy.go

```diff
@@ -160,7 +160,7 @@ func (lp *LoadPolicy) ListFrameworks() ([]string, error) {
 			continue
 		}
 
-		if contains(frameworkNames, framework.Name) {
+		if framework.Name == "" || contains(frameworkNames, framework.Name) {
```

Comment:
question: do we use this check as a substitute for a check that we did not manage to unmarshal a framework?

If I understand correctly, we assume that if the `Name == ""`, then the whole supposedly unmarshaled struct holds a zero value for each of its fields. Then, if the struct holds a zero value, we assume that it did not unmarshal correctly, so the contents of the file are not a framework definition, and therefore we skip it?

Is my understanding right?

Reply:
It is not a substitute: if the unmarshalling errors, then we skip earlier.
Checking the name is for cases when the descriptor has been unmarshalled without error but is obviously invalid (it is not a framework, even if the JSON may be fit in a framework struct).

However, unmarshalling errors are not a good rule to decide whether the file being loaded is indeed a framework descriptor. The name check works a little better in that respect, but again, it is not perfect.

Ideally, like previously discussed with David (@dwertent), we should have a "Kind" field in those files to clearly discriminate between the various descriptors.

Reply:
Yup, this is pretty much what I wanted to verify. Thank you!